RIOT-OS/RIOT
# make: boards: remove use of export with debug adapter related variables

### Contribution description

There's no need to export `DEBUG_ADAPTER`/`DEBUG_ADAPTER_ID`/`ST_LINK` variables since they are not passed to an external process.




### Testing procedure

- Verify that flashing a sam0 based board (samr21-xpro, samr30-xpro, etc) works. Try also by passing `DEBUG_ADAPTER_ID` with multiple boards plugged on the same computer
- Verify that flashing an st-link based board (one that is modified by this PR) works




### Issues/PRs references

Removal of unused exports https://github.com/RIOT-OS/RIOT/issues/10850
